compiler/resolution/cullOverReferences.cpp

```diff
@@ -185,54 +179,101 @@ void cullOverReferences() {
 
   buildDefUseMaps(defMap, useMap);
 
-  forv_Vec(CallExpr, call, gCallExprs) {
-    // Is this call to a non-primitive function?
-    if (FnSymbol* fn = call->isResolved()) {
-      // Does that function have a "value" counter part?
-      if (FnSymbol* valueFn = fn->valueFunction) {
-        if (CallExpr* move = toCallExpr(call->parentExpr)) {
-          INT_ASSERT(move->isPrimitive(PRIM_MOVE));
-
-          SymExpr* se = toSymExpr(move->get(1));
-
-          INT_ASSERT(se);
-
-          // Should we switch to the by-value form?
-          if (shouldUseByValueFunction(fn, se, defMap, useMap) == true) {
-            SET_LINENO(move);
-
-            SymExpr*   base = toSymExpr(call->baseExpr);
-            VarSymbol* tmp  = newTemp(valueFn->retType);
-
-            // Swap in the by-value implementation
-            base->var = valueFn;
-
-            // Generate a value temp to receive the value
-            move->insertBefore(new DefExpr(tmp));
-
-            if (requiresImplicitDestroy(call)) {
-              if (isString(valueFn->retType) == false) {
-                tmp->addFlag(FLAG_INSERT_AUTO_COPY);
-                tmp->addFlag(FLAG_INSERT_AUTO_DESTROY);
-              } else {
-                tmp->addFlag(FLAG_INSERT_AUTO_DESTROY);
-              }
-            }
-
-            if (useMap.get(se->var) && useMap.get(se->var)->n > 0) {
-              move->insertAfter(new CallExpr(PRIM_MOVE,
-                                             se->var,
-                                             new CallExpr(PRIM_ADDR_OF, tmp)));
-            } else {
-              se->var->defPoint->remove();
-            }
-
-            se->var = tmp;
+  forv_Vec(ContextCallExpr, cc, gContextCallExprs) {
+    CallExpr* refCall = cc->getRefCall();
+    CallExpr* valueCall = cc->getRValueCall();
+
+    INT_ASSERT(refCall);
+    INT_ASSERT(valueCall);
+
+    // Make sure that the context call only has 2 options.
+    INT_ASSERT(cc->options.length == 2);
+
```

Comment:
Optionally, replace the above with:

    INT_ASSERT(cc->options.length == 2);
    CallExpr* refCall = cc->getRefCall()->remove();
    CallExpr* valueCall = cc->getRValueCall()->remove();